pkg/forwarder/internal/retry/on_disk_retry_queue.go

```diff
@@ -95,13 +95,15 @@ func (s *onDiskRetryQueue) Store(transactions []transaction.Transaction) error {
 		_ = os.Remove(file.Name())
 		return err
 	}
-	defer file.Close()
-
 	s.currentSizeInBytes += bufferSize
 	s.filenames = append(s.filenames, file.Name())
 	s.telemetry.setFileSize(bufferSize)
 	s.telemetry.setCurrentSizeInBytes(s.GetDiskSpaceUsed())
 	s.telemetry.setFilesCount(s.getFilesCount())
+	err = file.Close()
+	if err != nil {
+		return fmt.Errorf("could not close file [%s]: %w", file.Name(), err)
```

Comment:
When we create the file or write to the file above, we return a plain error without the extra message, would it make sense to do the same here for consistency?

Reply:
It makes sense, I applied the suggestion.